css-ui-4/Overview.bs

```diff
@@ -2166,6 +2166,129 @@ Document languages should specify
 for each <a>widget</a> that they define
 what needs to be preserved.
 
+The possible kinds of widgets are ''appearance/none'', ''button'', ''searchfield'', ''textfield'',
+''slider-horizontal'', ''checkbox'', ''radio'', ''listbox'', ''menulist'', ''menulist-button'',
+''textarea'', ''progress-bar'', ''meter''.
+
+<div algorithm="compute kind of widget">
+    UAs must run this at [=used value=] time: compute the kind of widget to use for an element, |element|:
+
+    <ol>
+        <!-- none always works -->
+        <li>
+            If |computedAppearance| is ''appearance/none'', then return ''appearance: none''.
+
+        <li>
+            Let |computedAppearance| be the [=computed value=] of |element|'s 'appearance' property.
+
+        <li>
+            Let |authorProps| be a map of all properties that have a <a>cascaded value</a> for
+            |element|, where the [=Author Origin=], [=Animation Origin=] and [=Transition Origin=]
+            [=Cascade Origins=] are included and the [=User Origin=] and [=UA Origin=] [=cascade origins=]
+            are ignored, where the key is the property name and the value is that <a>cascaded value</a>.
+
+        <li>
+            If |element| can be a ''button'', given |computedAppearance|, then return the appropriate widget with
```

Comment:
My idea here is that "can be a button" should be a defined term in HTML, and linked from CSS UI.

Maybe the term needs to be defined in CSS UI itself, and it can say that it's up to the host language to define which elements "can be a button" (and in practice only HTML will do this).

Something like

```
Host languages can define which elements, if any, <dfn export>can be a button</dfn>, <dfn export>can be a textfield</dfn> ....

Note: HTML defines these in (link...).
```